src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,100 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newValue: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [movieTitle, setMovieTitle] = useState('');
+  const [movieDescription, setMovieDescription] = useState('');
+  const [movieImgUrl, setMovieImgUrl] = useState('');
+  const [movieImdbUrl, setMovieImdbUrl] = useState('');
+  const [movieImdbId, setMovieImdbId] = useState('');
+  const isEveryRequiredDataProvided = movieTitle
+    && movieImdbId && movieImdbUrl && movieImgUrl;
+  const newMovieData = {
+    title: movieTitle,
+    description: movieDescription,
+    imgUrl: movieImgUrl,
+    imdbUrl: movieImdbUrl,
+    imdbId: movieImdbId,
+  };
```

Comment:
You could name them as Movie property and 
```suggestion
  const newMovieData = {
    title,
    description,
    imgUrl,
    imdbUrl,
    imdbId,
  };
```